source/common/http/alternate_protocols_cache_impl.cc

```diff
@@ -174,38 +174,63 @@ std::chrono::microseconds AlternateProtocolsCacheImpl::getSrtt(const Origin& ori
   return entry_it->second.srtt;
 }
 
-void AlternateProtocolsCacheImpl::setAlternativesImpl(const Origin& origin,
-                                                      std::vector<AlternateProtocol>& protocols) {
-  static const size_t max_protocols = 10;
-  if (protocols.size() > max_protocols) {
-    ENVOY_LOG_MISC(trace, "Too many alternate protocols: {}, truncating", protocols.size());
-    protocols.erase(protocols.begin() + max_protocols, protocols.end());
+AlternateProtocolsCacheImpl::ProtocolsMap::iterator
+AlternateProtocolsCacheImpl::setAlternativesImpl(const Origin& origin,
```

Comment:
From the name of this method I would have assumed it would only be setting the alternative protocols but from the parameters and the code it's setting any of the fields. Should we consider renaming this to setImpl() or some such?

Reply:
I'm going with setPropertiesImpl for upcoming HttpServerProprerties name change